### PR TITLE
Implements handle login

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: src/ts/auth/AuthAPI.ts [hunk 1]
```diff
@@ -71,4 +66,29 @@ export default class AuthAPI {
...
+    if (this.isValidEmail(login.identifier)) {
+      data.email = login.identifier;
+    } else if (this.isValidHandle(login.identifier)) {
+      data.handle = login.identifier.replace('@', '');
+    }
+    return data;
+  }
+
+  private isValidEmail(email: string) {
+    const regExp = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (bennyn) [inline src/ts/auth/AuthAPI.ts]: I don't think that's  a good idea to have such an intense regex in a client API. Every login call would need to get through that check. The `generateCredentialData` expects already `LoginData` and the login data already has an `email`, `phone` or `handle` set. So there is no need to check again if an `email` or an `handle` should be used.

File: src/ts/auth/AuthAPI.ts [hunk 2]
```diff
@@ -71,4 +66,29 @@ export default class AuthAPI {
...
+    return regExp.test(email);
+  };
+
+  private isValidHandle(username: string) {
+    const regExp = /^[a-z_0-9]{2,21}$/;
```
Comment (bennyn) [inline src/ts/auth/AuthAPI.ts]: Do Android and iOS use the same regex? Doesn't the backend tell us on login if the handle is valid or not?

Comment (lipis) [inline src/ts/auth/AuthAPI.ts]: I totally agree.. we should skip these regexp altogether. Very basic one to see if the `@` is in the beginning or not and based on that figure out if it's email or username.

File: src/ts/auth/LoginData.ts [hunk 3]
```diff
@@ -1,7 +1,5 @@
 interface LoginData {
-  email: string;
-  phone: string;
+  identifier: string;
   password: number | string;
-  code: string;
```
Comment (bennyn) [inline src/ts/auth/LoginData.ts]: Why did you remove `phone` and `code`? These are valid attributes of the `Login` model!

**Screenshot**

unbenannt

File: src/ts/auth/AuthAPI.ts (hunk 2 above)
Comment (bennyn) [inline src/ts/auth/AuthAPI.ts]: No need to look for the `@`. Just check if `LoginData.email` or `LoginData.handle` is set. We can make easy / fancy / regex magic on the layer on top of the API layer (which would be an `AuthService` or sth. like that) to expose an easy-to-use library for external developers.

Comment (lipis) [inline src/ts/auth/AuthAPI.ts]:
```
isEmail(identifier) {
  return identifier.indexOf('@') > 0;
}
```

and if it's not `email` then it's `handle` and act accordignly.. backend will reply with the answer.

File: src/ts/auth/LoginData.ts [hunk 4]
```diff
@@ -1,7 +1,5 @@
 interface LoginData {
-  email: string;
-  phone: string;
+  identifier: string;
```
Comment (bennyn) [inline src/ts/auth/LoginData.ts]: There is no `identifier` within the `Login` model of the backend.

File: src/ts/auth/AuthAPI.ts [hunk 5]
```diff
@@ -19,12 +19,10 @@ export default class AuthAPI {
   }
 
   postCookiesRemove(login: LoginData, labels?: string[]): AxiosPromise {
+    const data: any = this.generateCredentialData(login);
```
Comment (bennyn) [inline src/ts/auth/AuthAPI.ts]: Why do you need `this.generateCredentialData`?

Can't you just do the following?

```javascript
if (login.email) {
  data.email = login.email;
}
```

Or simply: 

```javascript
data: login
```

File: README.md [hunk 6]
```diff
@@ -41,8 +41,11 @@ yarn add @wireapp/api-client
...
 
-IDENTIFIER = "name@email.com";
+EMAIL = "name@email.com";
+USERNAME = "username";
 PASSWORD = "secret";
 
-node dist/index.js --i="$IDENTIFIER" --p="$PASSWORD"
+node dist/index.js --e="$EMAIL" --p="$PASSWORD"
+# or
+node dist/index.js --h="$USERNAME" --p="$PASSWORD"
```
Comment (lipis) [inline README.md]: maybe rename everything to `HANDLE`? according to backend and only in the description somewhere mention (username) in the brackets
